src/Stratis.Bitcoin.Features.Wallet/WalletRPCController.cs

```diff
@@ -467,7 +467,13 @@ public async Task<object> GetTransaction(string txid, bool include_watchonly = f
                 throw new ArgumentException(nameof(txid));
 
             if (include_watchonly)
-                return await GetWatchOnlyTransactionAsync(trxid);
+            {
+                WalletHistoryModel history = await GetWatchOnlyTransactionAsync(trxid);
+                if ((history?.AccountsHistoryModel?.FirstOrDefault()?.TransactionsHistory?.Count ?? 0) == 0)
```

Comment:
I am not sure this is actually desirable behaviour, if the transaction is not present in the watch-only account then it may still be present in the regular wallet. Perhaps inverting the check and only returning early if there is in fact a history record available would be best.

Reply:
Good point. I agree.